lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -528,22 +528,25 @@ static bool sinkCode(Function *F) {
 
     // Sink RELU below batch concat nodes.
     if (auto *CN = dyn_cast<ConcatNode>(node)) {
-      if (CN->getInputs().size() != 2) {
-        continue;
+      llvm::SmallVector<NodeValue, 6> CNInputs;
+      bool isAllRelu = true;
+      for (auto input : CN->getInputs()) {
+        auto *inputRL = dyn_cast<ReluNode>(input);
+        if (!inputRL) {
+          isAllRelu = false;
+          break;
+        }
+        CNInputs.push_back(inputRL->getInput());
       }
-      auto LInput = CN->getInputs()[0];
-      auto RInput = CN->getInputs()[1];
-      auto *L = dyn_cast<ReluNode>(LInput);
-      auto *R = dyn_cast<ReluNode>(RInput);
 
-      if (L && R) {
-        auto *newCN = F->createConcat(
-            CN->getName(), {L->getInput(), R->getInput()}, CN->getDim());
+      if (isAllRelu) {
+        auto *newCN = F->createConcat(CN->getName(), CNInputs, CN->getDim());
         newCN->setPredicate(node->getPredicate());
-        auto *newRL =
-            F->createRELU(L->getName(), newCN, CN->getResult().getType());
+        auto name = dyn_cast<ReluNode>(CN->getNthInput(0))->getName();
```

Comment:
nit: you don't need to cast it to `ReluNode` to get its name -- can just do `CN->getNthInput(0).getName()`

Reply:
getNthInput returns NodeValue which doesn't have getName().
Did you mean something like: CN->getNthInput(0).getNode()->getName()?

Reply:
Yep, sorry, guess there isn't forwarding there from NodeValue to name